src/sign/index.ts

```diff
@@ -52,13 +53,14 @@ export default class Client {
   async sign(web3: Web3Provider | Wallet, address: string, message, types) {
     // @ts-ignore
     const signer = web3?.getSigner ? web3.getSigner() : web3;
-    if (!message.from) message.from = address;
+    const checksumAddress = getAddress(address.toLowerCase());
+    if (!message.from) message.from = checksumAddress;
```

Comment:
I feel we should do checksum for `message.from` too.

Reply:
Of course

Reply:
This checksum message.from only if its not part of the message, if its part of the message it wont be checksumed you should change that